app/controllers/peer_reviews_controller.rb

```diff
@@ -1,65 +1,183 @@
 class PeerReviewsController < ApplicationController
+  include GroupsHelper
+
   before_action :set_peer_review, only: [:show, :edit, :update, :destroy]
   before_filter :authorize_only_for_admin
 
-  # GET /peer_reviews
-  # GET /peer_reviews.json
   def index
-    @peer_reviews = PeerReview.all
+    @assignment = Assignment.find(params[:assignment_id])
+    @section_column = ''
+    if Section.all.size > 0
+      @section_column = "{
+        id: 'section',
+        content: '" + t(:'summaries_index.section') + "',
+        sortable: true
+      },"
+    end
   end
 
-  # GET /peer_reviews/1
-  # GET /peer_reviews/1.json
-  def show
+  def populate
+    @assignment = Assignment.find(params[:assignment_id])
+
+    reviewer_groups = get_groupings_table_info()
+    reviewee_groups = get_groupings_table_info(@assignment.parent_assignment)
+
+    reviewee_to_reviewers_map = create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    id_to_group_names_map = create_map_group_id_to_name(reviewer_groups, reviewee_groups)
+    num_reviews_map = create_map_number_of_reviews_for_reviewer(reviewer_groups)
```

Comment:
I'm unhappy with the functions above and feel they are too verbose, I'll look into a way to reduce my logic into a single more appropriate data structure.